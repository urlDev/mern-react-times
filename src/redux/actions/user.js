import axios from 'axios';
import toaster from 'toasted-notes';
import { push } from 'connected-react-router';

import NotificationComponent from '../../components/notification-component/NotificationComponent';

const url = process.env.REACT_APP_API_URL;

export const REGISTER_USER = 'REGISTER_USER';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGIN_USER = 'LOGIN_USER';
export const DELETE_USER = 'DELETE_USER';
export const UPDATE_USER = 'UPDATE_USER';
export const GET_USER = 'GET_USER';
export const SET_TOKEN = 'SET_TOKEN';
export const USER_FETCH_ERROR = 'USER_FETCH_ERROR';
export const USER_LOADING = 'USER_LOADING';
export const USER_LOADING_END = 'USER_LOADING_END';
export const USER_MODAL_OPEN = 'USER_MODAL_OPEN';
export const USER_MODAL_CLOSE = 'USER_MODAL_CLOSE';
export const DELETE_MODAL_OPEN = 'DELETE_MODAL_OPEN';
export const DELETE_MODAL_CLOSE = 'DELETE_MODAL_CLOSE';
export const UPLOAD_AVATAR = 'UPLOAD_AVATAR';

export const registerUser = (user) => ({
  type: REGISTER_USER,
  payload: user,
});

export const logOutUser = () => ({
  type: LOGOUT_USER,
});

export const loginUser = (user) => ({
  type: LOGIN_USER,
  payload: user,
});

export const deleteUser = () => ({
  type: DELETE_USER,
});

export const updateUser = (user) => ({
  type: UPDATE_USER,
  payload: user,
});

export const getUser = (user) => ({
  type: GET_USER,
  payload: user,
});

export const setToken = (token) => ({
  type: SET_TOKEN,
  payload: token,
});

export const userFetchError = (error) => ({
  type: USER_FETCH_ERROR,
  payload: error,
});

export const userLoading = () => ({
  type: USER_LOADING,
});

export const userLoadingEnd = () => ({
  type: USER_LOADING_END,
});

export const userModalOpen = () => ({
  type: USER_MODAL_OPEN,
});

export const userModalClose = () => ({
  type: USER_MODAL_CLOSE,
});

export const deleteModalOpen = () => ({
  type: DELETE_MODAL_OPEN,
});

export const deleteModalClose = () => ({
  type: DELETE_MODAL_CLOSE,
});

export const uploadAvatar = (images) => ({
  type: UPLOAD_AVATAR,
  payload: images,
});

export const fetchRegisterUser = (user) => async (dispatch) => {
  dispatch(userLoading());
  try {
    const response = await axios.post(`${url}/profile/register`, user);
    const data = await response.data;
    return [
      dispatch(registerUser(data.user)),
      dispatch(userLoadingEnd()),
      dispatch(setToken(data.token)),
      localStorage.setItem('token', JSON.stringify(data.token)),
      localStorage.setItem('user', JSON.stringify(data.user)),
      dispatch(push('/')),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`Welcome! Enjoy your time`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      dispatch(userLoadingEnd()),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

export const fetchLogoutUser = () => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };

  const path = window.location.pathname;

  try {
    await axios.post(`${url}/profile/logout`, null, config);

    return [
      path.includes('profile') && dispatch(push('/home')),
      dispatch(logOutUser()),
      toaster.notify(
        () => <NotificationComponent text={'Buh-Bye!'} success={true} />,
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return dispatch(userFetchError(error));
  }
};

export const fetchLoginUser = (user) => async (dispatch) => {
  dispatch(userLoading());
  try {
    const response = await axios.post(`${url}/profile/login`, user);
    const data = await response.data;
    return [
      dispatch(loginUser(data.user)),
      dispatch(userLoadingEnd()),
      dispatch(setToken(data.token)),
      localStorage.setItem('token', JSON.stringify(data.token)),
      localStorage.setItem('user', JSON.stringify(data.user)),
      dispatch(push('/home')),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`Welcome! Enjoy your time`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      dispatch(userLoadingEnd()),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

// To get the user,
// but because I added an or statement in user and token initial states,
// app will check if there is user and token in localStorage
// so there will be no need for this.
// Adding just in case.
export const fetchUser = () => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  dispatch(userLoading());
  try {
    const response = await axios.get(`${url}/profile`, config);
    const data = await response.data;
    return [
      dispatch(registerUser(data.user)),
      dispatch(userLoadingEnd()),
      dispatch(setToken(data.token)),
      localStorage.setItem('token', JSON.stringify(data.token)),
      localStorage.setItem('user', JSON.stringify(data.user)),
    ];
  } catch (error) {
    return [dispatch(userFetchError(error)), dispatch(userLoadingEnd())];
  }
};

export const fetchUpdateUser = (user) => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  try {
    const response = await axios.patch(`${url}/profile`, user, config);
    const data = await response.data;
    return [
      dispatch(updateUser(data)),
      localStorage.setItem('user', JSON.stringify(data)),
      dispatch(push('/')),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`Updated successfully!`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

export const fetchDeleteUser = () => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  try {
    await axios.delete(`${url}/profile`, config);
    return [
      localStorage.clear(),
      dispatch(push('/')),
      dispatch(deleteUser()),
      toaster.notify(
        () => <NotificationComponent success={true} text={`Buh-bye!`} />,
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            success={false}
            text={`Something went wrong!`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

export const fetchUploadAvatar = (input) => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };

  try {
    const response = await axios.post(`${url}/profile/avatar`, input, config);
    const data = await response.data;
    return [
      dispatch(uploadAvatar(data)),
      localStorage.setItem('user', JSON.stringify(data)),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Avatar is changed! Wow, that looks amazing! ;)'}
            success={true}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};
