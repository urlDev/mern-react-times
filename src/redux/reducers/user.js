import {
    REGISTER_USER,
    SET_TOKEN,
    USER_FETCH_ERROR,
    USER_LOADING,
    USER_LOADING_END,
} from '../actions/user';

const initialState = {
    loadingUser: false,
    errorUser: null,
    user: JSON.parse(localStorage.getItem('user')) || {},
    token: JSON.parse(localStorage.getItem('token')) || {},
};

const userReducer = (state = initialState, action) => {
    switch (action.payload) {
        case REGISTER_USER:
            return {
                ...state,
                user: action.payload,
            };
        case SET_TOKEN:
            return {
                ...state,
                token: action.payload,
            };
        case USER_FETCH_ERROR:
            return {
                ...state,
                error: action.payload,
            };
        case USER_LOADING:
            return {
                ...state,
                loading: true,
            };
        case USER_LOADING_END:
            return {
                ...state,
                loading: false,
            };
        default:
            return state;
    }
};

export default userReducer;