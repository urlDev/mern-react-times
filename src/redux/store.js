import { createStore, applyMiddleware } from 'redux';
import rootReducer from './rootReducer';

import thunk from 'redux-thunk';
import logger from 'redux-logger';

const middlewares = [thunk, logger];

export default createStore(rootReducer, applyMiddleware(...middlewares));