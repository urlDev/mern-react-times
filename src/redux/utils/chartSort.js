const chartSort = (prevState, newState) => {
    const tempState = [...prevState, [...newState]];

    let sortedState = [...tempState];

    sortedState = sortedState
        .sort((a, b) => a[1] - b[1])
        .map((stock) => stock.slice(0, 1));

    return sortedState;
};

export default chartSort;