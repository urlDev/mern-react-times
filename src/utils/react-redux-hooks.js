// Mocking redux hooks
import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from "react-redux";

export const useSelector = (state) => originalUseSelector(state);
export const useDispatch = () => originalUseDispatch();
