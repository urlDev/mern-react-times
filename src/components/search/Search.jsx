import React from 'react';
import { useDispatch } from 'react-redux';

import { fetchSearch } from '../../redux/actions/chart';

import SearchModal from '../search-modal/SearchModal';
import SearchSrc from '../../assets/searchIcon.svg';
import { NavSearchContainer, SearchIcon } from './Search.styles';

const Search = ({ icon }) => {
  const [open, setOpen] = React.useState(false);
  const [input, setInput] = React.useState('');
  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(fetchSearch(input));
    setInput('');
  };

  return (
    <NavSearchContainer open={open}>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          placeholder="Search for a company or symbol"
          onChange={(e) => {
            setInput(e.target.value);
            dispatch(fetchSearch(e.target.value));
          }}
          open={open}
        />
        <SearchIcon icon={icon} open={open} onClick={() => setOpen(!open)}>
          <img src={SearchSrc} alt="search icon" />
        </SearchIcon>
      </form>
      {open ? <SearchModal /> : null}
    </NavSearchContainer>
  );
};

export default Search;
