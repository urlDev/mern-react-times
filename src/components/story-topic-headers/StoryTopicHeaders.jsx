import moment from "moment";
import { useDispatch, useSelector } from "react-redux";

import { changeHeader, clearError } from "../../redux/actions/news";

import ResponsiveMenu from "../responsive-menu/ResponsiveMenu";
import ResponsiveMenuModal from "../responsive-menu-modal/ResponsiveMenuModal";

import {
  StoryTopicHeadersContainer,
  TopicContainer,
  Date,
  StyledLink,
} from "./StoryTopicHeaders.styles";

export const topics = [
  "Home",
  "Arts",
  "Business",
  "Health",
  "Opinion",
  "Politics",
  "Science",
  "Sports",
  "Technology",
  "Travel",
];

const StoryTopics = () => {
  const { width, responsiveMenu } = useSelector((news) => news.news);
  const dispatch = useDispatch();

  return (
    <StoryTopicHeadersContainer>
      <Date>
        <span>{moment().format("dddd | ")}</span>
        {moment().format("MMM DD, YYYY")}
      </Date>
      <TopicContainer>
        {width < 1024 ? (
          <>
            <ResponsiveMenu />
            {responsiveMenu ? <ResponsiveMenuModal /> : null}
          </> //
        ) : (
          <>
            {topics.map((topic) => {
              return (
                <StyledLink key={topic} to={`/${topic.toLowerCase()}`}>
                  <span
                    onClick={() => {
                      dispatch(changeHeader(topic.toLowerCase()));
                      dispatch(clearError());
                    }}
                  >
                    {topic}
                  </span>
                </StyledLink>
              );
            })}
          </> //
        )}
      </TopicContainer>
    </StoryTopicHeadersContainer>
  );
};

export default StoryTopics;
