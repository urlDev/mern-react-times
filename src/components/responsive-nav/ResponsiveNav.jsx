import { useSelector } from "utils/react-redux-hooks";

import Nav from "components/nav/Nav";
import Search from "components/search/Search";
import StoryTopicHeaders from "components/story-topic-headers/StoryTopicHeaders";

import LogoDarkSrc from "assets/logo.png";

const ResponsiveNav = () => {
  const { width } = useSelector((state) => state.news);
  return (
    <>
      {width < 768 ? (
        <>
          <Nav
            logo={LogoDarkSrc}
            borderBottom="1px solid lightgray"
            icon="block"
          />
          <Search icon="block" />
        </> //
      ) : (
        <>
          <Nav
            logo={LogoDarkSrc}
            borderBottom="1px solid lightgray"
            icon="block"
          />
          <StoryTopicHeaders icon="block" />
        </> //
      )}
    </> //
  );
};

export default ResponsiveNav;
