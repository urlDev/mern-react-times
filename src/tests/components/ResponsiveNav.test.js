import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";

import * as ReactReduxHooks from "utils/react-redux-hooks";
import { mockStore } from "tests/store";
import { user } from "tests/fixtures/user";

import Search from "components/search/Search";
import StoryTopicHeaders from "components/story-topic-headers/StoryTopicHeaders";
import ResponsiveNav from "components/responsive-nav/ResponsiveNav";

describe("Testing ResponsiveNav component", () => {
  let wrapper;
  let store;

  beforeEach(() => {
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation((state) => store.getState());

    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);
  });

  describe("Testing for when the width is smaller than 768", () => {
    beforeEach(() => {
      store = mockStore({
        width: 700,
        user,
      });

      wrapper = mount(
        <Router>
          <ResponsiveNav store={store} />
        </Router>
      );
    });

    test("Should render the component with Nav and Search", () => {
      expect(wrapper).toHaveLength(1);

      expect(wrapper.containsMatchingElement(<Search />)).toBeTruthy();
    });
    test("Should match the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Testing the component for when the width is bigger than 768", () => {
    beforeEach(() => {
      store = mockStore({
        user,
        width: 900,
      });

      wrapper = mount(
        <Router>
          <ResponsiveNav store={store} />
        </Router>
      );
    });

    test("Should render the component with Nav and StoryTopicHeaders", () => {
      expect(wrapper).toHaveLength(1);

      expect(
        wrapper.containsMatchingElement(<StoryTopicHeaders />)
      ).toBeTruthy();
    });
    test("Should match the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
