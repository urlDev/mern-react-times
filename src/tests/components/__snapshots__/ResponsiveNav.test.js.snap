// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ResponsiveNav component Testing for when the width is smaller than 768 Should match the snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ResponsiveNav
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Nav
        borderBottom="1px solid lightgray"
        icon="block"
        logo="logo.png"
      >
        <styled.nav
          borderBottom="1px solid lightgray"
        >
          <nav
            className="sc-bkzZxe gEOCrV"
          >
            <StoryTopicHeaders
              icon="block"
            >
              <styled.div>
                <div
                  className="sc-iBPRYJ kSTahC"
                >
                  <styled.p>
                    <p
                      className="sc-pFZIQ cGvNpk"
                    >
                      <span>
                        Saturday | 
                      </span>
                      Jan 16, 2021
                    </p>
                  </styled.p>
                  <styled.div>
                    <div
                      className="sc-fubCfw tAnSQ"
                    >
                      <ResponsiveMenu
                        icon="block"
                      >
                        <styled.div
                          alt="hamburger menu"
                          icon="block"
                          onClick={[Function]}
                        >
                          <div
                            alt="hamburger menu"
                            className="sc-gKsewC gMUjqL"
                            onClick={[Function]}
                          >
                            <img
                              alt="hamburger icon"
                              src="hamburger.svg"
                            />
                          </div>
                        </styled.div>
                      </ResponsiveMenu>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </StoryTopicHeaders>
            <LogoComponent
              logo="logo.png"
            >
              <Styled(Link)
                to="/home"
              >
                <Link
                  className="sc-crrsfI GrSVm"
                  to="/home"
                >
                  <LinkAnchor
                    className="sc-crrsfI GrSVm"
                    href="/home"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-crrsfI GrSVm"
                      href="/home"
                      onClick={[Function]}
                    >
                      <img
                        alt="react times logo"
                        onClick={[Function]}
                        src="logo.png"
                        style={
                          Object {
                            "height": "16px",
                          }
                        }
                      />
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </LogoComponent>
            <NavUserImage>
              <styled.div>
                <div
                  className="sc-dIUggk khhrJB"
                >
                  <styled.button
                    onClick={[Function]}
                  >
                    <button
                      className="sc-hHftDr bKLrHh"
                      onClick={[Function]}
                    >
                      <UserImage
                        border="29px"
                        margin="10px"
                      >
                        <styled.picture
                          border="29px"
                          margin="10px"
                        >
                          <picture
                            className="sc-fFubgz cqeoaP"
                          >
                            <source
                              srcSet="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzbMAAAACX"
                              type="image/png"
                            />
                            <img
                              alt="avatar"
                              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzbMAAAACX"
                            />
                          </picture>
                        </styled.picture>
                      </UserImage>
                    </button>
                  </styled.button>
                </div>
              </styled.div>
            </NavUserImage>
          </nav>
        </styled.nav>
      </Nav>
      <Search
        icon="block"
      >
        <styled.div>
          <div
            className="sc-eCssSg bIrBTP"
          >
            <form
              onClick={[Function]}
              onSubmit={[Function]}
            >
              <styled.input
                onChange={[Function]}
                placeholder="Search for a company or symbol"
                type="text"
                value=""
              >
                <input
                  className="sc-jSgupP ibTSId"
                  onChange={[Function]}
                  placeholder="Search for a company or symbol"
                  type="text"
                  value=""
                />
              </styled.input>
              <styled.a
                icon="block"
                onClick={[Function]}
              >
                <a
                  className="sc-hKgILt itWKZn"
                  onClick={[Function]}
                >
                  <img
                    alt="search icon"
                    src="searchIcon.svg"
                  />
                </a>
              </styled.a>
            </form>
            <SearchModal />
          </div>
        </styled.div>
      </Search>
    </ResponsiveNav>
  </Router>
</BrowserRouter>
`;

exports[`Testing ResponsiveNav component Testing the component for when the width is bigger than 768 Should match the snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ResponsiveNav
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Nav
        borderBottom="1px solid lightgray"
        icon="block"
        logo="logo.png"
      >
        <styled.nav
          borderBottom="1px solid lightgray"
        >
          <nav
            className="sc-bkzZxe gEOCrV"
          >
            <Search
              icon="block"
            >
              <styled.div>
                <div
                  className="sc-eCssSg bIrBTP"
                >
                  <form
                    onClick={null}
                    onSubmit={[Function]}
                  >
                    <styled.input
                      onChange={[Function]}
                      placeholder="Search for a company or symbol"
                      type="text"
                      value=""
                    >
                      <input
                        className="sc-jSgupP ibTSId"
                        onChange={[Function]}
                        placeholder="Search for a company or symbol"
                        type="text"
                        value=""
                      />
                    </styled.input>
                    <styled.a
                      icon="block"
                      onClick={[Function]}
                    >
                      <a
                        className="sc-hKgILt itWKZn"
                        onClick={[Function]}
                      >
                        <img
                          alt="search icon"
                          src="searchIcon.svg"
                        />
                      </a>
                    </styled.a>
                  </form>
                  <SearchModal />
                </div>
              </styled.div>
            </Search>
            <LogoComponent
              logo="logo.png"
            >
              <Styled(Link)
                to="/home"
              >
                <Link
                  className="sc-crrsfI GrSVm"
                  to="/home"
                >
                  <LinkAnchor
                    className="sc-crrsfI GrSVm"
                    href="/home"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-crrsfI GrSVm"
                      href="/home"
                      onClick={[Function]}
                    >
                      <img
                        alt="react times logo"
                        onClick={[Function]}
                        src="logo.png"
                        style={
                          Object {
                            "height": "16px",
                          }
                        }
                      />
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </LogoComponent>
            <NavUserImage>
              <styled.div>
                <div
                  className="sc-dIUggk khhrJB"
                >
                  <styled.button
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="sc-idOhPF cedOKG"
                      onClick={[Function]}
                      type="button"
                    >
                      Logout
                    </button>
                  </styled.button>
                  <styled.button
                    onClick={[Function]}
                  >
                    <button
                      className="sc-hHftDr bKLrHh"
                      onClick={[Function]}
                    >
                      <UserImage
                        border="29px"
                        margin="10px"
                      >
                        <styled.picture
                          border="29px"
                          margin="10px"
                        >
                          <picture
                            className="sc-fFubgz cqeoaP"
                          >
                            <source
                              srcSet="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzbMAAAACX"
                              type="image/png"
                            />
                            <img
                              alt="avatar"
                              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzbMAAAACX"
                            />
                          </picture>
                        </styled.picture>
                      </UserImage>
                    </button>
                  </styled.button>
                </div>
              </styled.div>
            </NavUserImage>
          </nav>
        </styled.nav>
      </Nav>
      <StoryTopicHeaders
        icon="block"
      >
        <styled.div>
          <div
            className="sc-iBPRYJ kSTahC"
          >
            <styled.p>
              <p
                className="sc-pFZIQ cGvNpk"
              >
                <span>
                  Saturday | 
                </span>
                Jan 16, 2021
              </p>
            </styled.p>
            <styled.div>
              <div
                className="sc-fubCfw tAnSQ"
              >
                <ResponsiveMenu
                  icon="block"
                >
                  <styled.div
                    alt="hamburger menu"
                    icon="block"
                    onClick={[Function]}
                  >
                    <div
                      alt="hamburger menu"
                      className="sc-gKsewC gMUjqL"
                      onClick={[Function]}
                    >
                      <img
                        alt="hamburger icon"
                        src="hamburger.svg"
                      />
                    </div>
                  </styled.div>
                </ResponsiveMenu>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </StoryTopicHeaders>
    </ResponsiveNav>
  </Router>
</BrowserRouter>
`;
